package/MDAnalysis/core/topology.py

```diff
@@ -17,13 +17,58 @@
 Topology object --- :mod:`MDAnalysis.core.topology'
 ===================================================================
 
+
+"""
+
+"""
+# second docstring == multiline comment
+
+TODO Notes:
+  Could make downshift tables lazily built! This would
+    a) Make these not get built when not used
+    b) Optimise moving multiple atoms between residues as only built once afterwards
+
+  Could optimise moves by only updating the two parent tables rather than rebuilding everything!
+
+
 """
+from six.moves import zip
 import numpy as np
 
 from ..lib.mdamath import one_to_many_pointers
 from .topologyattrs import Atomindices, Resindices, Segindices
 
 
+def make_downshift_arrays(upshift):
+    """From an upwards translation table, create the opposite direction
+
+    Turns a many to one mapping to a one to many mapping
+
+    Eg :: 
+      AR = np.array([0, 1, 0, 2, 2, 0, 2])
+    
+      make_downshift_arrays(AR)
+      array([array([0, 2, 5]), array([1]), array([3, 4, 6])], dtype=object)
```

Comment:
please be consistent with the numpy docs. and what the hell does this function do? I can't make out any pattern between input and output.